src/smt/theory_str_noodler/theory_str_noodler_final_check.cpp

```diff
@@ -784,18 +784,21 @@ namespace smt::noodler {
 
     lbool theory_str_noodler::run_length_proc(const Formula& instance, const AutAssignment& aut_assignment, const std::unordered_set<BasicTerm>& init_length_sensitive_vars) {
         STRACE("str", tout << "Trying length-based procedure" << std::endl);
-        LengthDecisionProcedure nproc(instance, aut_assignment, init_length_sensitive_vars, m_params);
+        // we need a method get_formula from LengthDecisionProcedure
```

Comment:
I do not understand this comment. Is it saying that we need to keep shared pointer to `LengthDecisionProcedure` instead of `LengthDecisionProcedure`?

Reply:
Yes, I am using here a method, which is not part of `AbstractDecisionProcedure`